tests/app/service/test_rest.py

```diff
@@ -1881,8 +1881,45 @@ def test_get_notifications_for_service_without_page_count(
         count_pages=False
     )
     assert len(resp['notifications']) == 1
-    assert resp['total'] is None
     assert resp['notifications'][0]['id'] == str(without_job.id)
+    assert 'prev' not in resp['links']
+    assert 'next' not in resp['links']
+
+
+def test_get_notifications_for_service_pagination_links(
```

Comment:
Do we want a test for `count_pages = False`?

Reply:
I'm leaning towards no that would be testing the internal implementation, rather than the interface. I generally try and avoid tests of the internal implementation but happy to be convinced or told that this is something you see as essential